ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/MinimalState.java

```diff
@@ -268,39 +301,23 @@ protected boolean initializePlatformProperties() {
 	static {
 		// Listen to changes in the available VMInstalls and
 		// ExecutionEnvironment defaults
-		@SuppressWarnings("restriction")
-		String nodeQualifier = org.eclipse.jdt.internal.launching.LaunchingPlugin.ID_PLUGIN;
-		IEclipsePreferences launchingNode = InstanceScope.INSTANCE.getNode(nodeQualifier);
-		launchingNode.addPreferenceChangeListener(e -> {
-			if (e.getKey().equals("org.eclipse.jdt.launching.PREF_DEFAULT_ENVIRONMENTS_XML")) { //$NON-NLS-1$
-				Object oldValue = e.getOldValue() == null ? "" : e.getOldValue(); //$NON-NLS-1$
-				Object newValue = e.getNewValue() == null ? "" : e.getNewValue(); //$NON-NLS-1$
-				if (!oldValue.equals(newValue)) {
-					triggerSystemPackagesReload();
-				}
-			}
-		});
-		JavaRuntime.addVMInstallChangedListener(new IVMInstallChangedListener() {
-			@Override
-			public void vmRemoved(IVMInstall vm) {
-				triggerSystemPackagesReload();
-			}
-
-			@Override
-			public void vmChanged(PropertyChangeEvent event) {
-				triggerSystemPackagesReload();
-			}
+		IEclipsePreferences launchingNode = getJdtLaunchingPreferences();
+		launchingNode.addPreferenceChangeListener(PREF_CHANGE_LISTENER);
+		JavaRuntime.addVMInstallChangedListener(VM_CHANGED_LISTENER);
+	}
 
-			@Override
-			public void vmAdded(IVMInstall vm) {
-				triggerSystemPackagesReload();
-			}
+	public static void shutdown() {
```

Comment:
No need to have this public.
```suggestion
	static void shutdown() {
```